### PR TITLE
PCRM-3724 - Update retry feature; apply demo fixes

### PR DESCRIPTION
**Pull request by jamesnunnnhs**

### CODE REVIEW

File: packages/lambda_code/LR_12_pds_registration_status/lr_12_lambda_handler.py [hunk 1]
```diff
@@ -109,19 +109,21 @@ def get_pds_exclusive_registrations(self, job_id: str) -> Dict[str, str]:
...
+                            marker.value == pds_record.get("sensitive")
+                            for marker in SensitiveMarkers
+                        ):
+                            self.log_object.write_log(
+                                "UTI9995",
+                                None,
+                                {
+                                    "logger": "LR12.Lambda",
+                                    "level": "INFO",
+                                    "message": "Sensitive patient, skipping registration",
```
Comment (josh-scargill-nhsd) [inline packages/lambda_code/LR_12_pds_registration_status/lr_12_lambda_handler.py:123]: We'd want the patient id here surely?

Comment (josh-scargill-nhsd) [reply packages/lambda_code/LR_12_pds_registration_status/lr_12_lambda_handler.py]: and jobId

**Review by josh-scargill-nhsd — APPROVED** [review]

File: packages/lambda_code/LR_12_pds_registration_status/lr_12_lambda_handler.py (hunk 1 above)
Comment (jamesnunnnhs) [reply packages/lambda_code/LR_12_pds_registration_status/lr_12_lambda_handler.py]: Ok. FYI this is pds-exclusive patients which won't have a PatientId so NHS number is the only other id we could report.

**Review by abbas-khan8 — DISMISSED** [review]

**Review by HarveyNHS — DISMISSED** [review]
Approved

**Review by josh-scargill-nhsd — DISMISSED** [review]

**Review by joeds13 — APPROVED** [review]

**Review by HarveyNHS — APPROVED** [review]
Approved

**Review by premnhs — APPROVED** [review]

**Pull request merged**